prow/plugins/trigger/trigger.go

```diff
@@ -322,3 +327,33 @@ func getPostsubmits(log *logrus.Entry, gc git.ClientFactory, cfg *config.Config,
 	}
 	return postsubmits
 }
+
+// createWithRetry will retry the cration of a ProwJob. The Name must be set, otherwise we might end up creating it multiple times
```

Comment:
do we need to check for the name to be set?

Reply:
I think it's fine, we don't have any code that uses GenerateName today but use the UUID lib for that everywhere.
/hold cancel